modules/nextflow/src/main/groovy/nextflow/k8s/K8sTaskHandler.groovy

```diff
@@ -462,9 +486,11 @@ class K8sTaskHandler extends TaskHandler implements FusionAwareTask {
         
         if( podName ) {
             log.trace "[K8s] deleting ${resourceType.lower()} name=$podName"
-            if ( useJobResource() )
+            if ( useJobResource() ) {
                 client.jobDelete(podName)
-            else
+                if ( resourceType == ResourceType.MPIJob )
+                    client.mpiJobDelete(podName.minus("-launcher"))
```

Comment:
@xhejtman trying to understand this code... does this code mean that an MPIJob creates a separate Job for the launcher, rather than a pod? Also, does deleting the MPIJob not also delete the underlying pods / jobs?

Reply:
yes, it creates a job for launcher with hardcoded suffix `-launcher`

Reply:
Just for a heads up - the Flux Operator doesn't have/need a launcher node.

Reply:
but does it not delete the launcher automatically when you delete the mpi job?